compose/src/main/java/com/bottlerocketstudios/compose/snippets/snippetDetails/SnippetDetailsUiModel.kt

```diff
@@ -0,0 +1,19 @@
+package com.bottlerocketstudios.compose.snippets.snippetDetails
+
+import com.bottlerocketstudios.brarchitecture.domain.models.Links
+import com.bottlerocketstudios.brarchitecture.domain.models.SnippetDetailsFile
+import com.bottlerocketstudios.brarchitecture.domain.models.User
+
+data class SnippetDetailsUiModel(
+    val id: String? = null,
```

Comment:
@MMurrayAtBR When creating these Domain/Ui model classes, it's a good idea to "clean up" null types where possible to prevent unnecessary null handling from all consumers of this data.

For example, the following properties could be non-nullable Strings that use the empty String as the default value:

* id
* title
* createdMessage
* updatedMessage
* httpsCloneLink
* sshCloneLink

A null String or "" usually mean the same thing. If the type was `String?`, then you might do something like `foo.isNullOrEmpty()` or `foo.isBlankOrEmpty()`. If the type was `String`, then you'd just need to check `foo.isEmpty()` or `foo.isBlank()`

Additionally, if all of the other fields are required per the endpoint response, then they could be made non-null here in the domain/ui model.

cc: @BottleRocket-Colin